### PR TITLE
Issue 240 - REST APIs for validating and creating workspaces

### PR DESCRIPTION
**Pull request by wardcr**
- Added 2 new views for creating and validating workspaces.
- Added a TODO in storage/models.py for how to handle broker-specific warnings based on the outcome of the NFS broker changes.
- Added a TODO in storage/test_views.py for testing a validation warning when changing the broker type. I had to temporarily hack around the fact that a 2nd broker (nfs) doesn't exist yet.

### CODE REVIEW

File: scale/docs/rest/workspace.rst [hunk 1]
```diff
@@ -2,181 +2,328 @@
...
+| Creates a new workspace with associated configuration                                                                   |
++-------------------------------------------------------------------------------------------------------------------------+
+| **POST** /workspaces/                                                                                                   |
++-------------------------+-----------------------------------------------------------------------------------------------+
+| **Content Type**        | *application/json*                                                                            |
++-------------------------+-----------------------------------------------------------------------------------------------+
+| **JSON Fields**                                                                                                         |
++-------------------------+-------------------+----------+----------------------------------------------------------------+
+| name                    | String            | Required | The stable name of the workspace used for queries.             |
```
Comment (tclarke) [inline scale/docs/rest/workspace.rst]: It's not entirely obvious what "stable name" is here. Maybe something like "identifying name" or something?

File: scale/docs/rest/workspace.rst [hunk 2]
```diff
@@ -2,181 +2,328 @@
...
+|    {                                                                                                                    |
+|        "name": "raw",                                                                                                   |
+|        "title": "Raw Source",                                                                                           |
+|        "description": "Raw Source Workspace",                                                                           |
+|        "base_url": "http://host.com/rs",                                                                                |
+|        "is_active": true,                                                                                               |
+|        "json_config": {                                                                                                 |
+|            "broker": {                                                                                                  |
+|                "type": "host"                                                                                           |
```
Comment (tclarke) [inline scale/docs/rest/workspace.rst]: There's additional information needed for a host broker. We should have a page describing all the available brokers and their broker specific configuration.

**Comment by wardcr** [comment]
TODO comments are now resolved after merging in the broker changes.

**Comment by dfaller** [comment]
LGTM

**Pull request merged**